services/motion/builtin/builtin.go

```diff
@@ -237,7 +270,118 @@ func (ms *builtIn) MoveOnGlobe(
+	if !ok {
+		return false, fmt.Errorf("cannot assert local of type %T as a movementSensor", local)
+	}
+
+	currentPosition, _, err := movementSensor.Position(ctx, nil)
+	if err != nil {
+		return false, err
+	}
+	currentPose := spatialmath.GeoPointToPose(currentPosition)
+
+	// convert destination into a spatialmath pose in frame with respect to 0 latitude, 0 longitude
+	dstPose := spatialmath.GeoPointToPose(destination)
+	dstPIF := referenceframe.NewPoseInFrame(referenceframe.World, dstPose)
+
+	// convert GeoObstacles into GeometriesInFrame then into a Worldstate
+	geoms := spatialmath.GeoObstaclesToGeometries(obstacles)
+	gif := referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)
+	wrldst, err := referenceframe.NewWorldState([]*referenceframe.GeometriesInFrame{gif}, nil)
+	if err != nil {
+		return false, err
+	}
+	// get worldstate limits
+	limits := wrldst.BoundingBox(ctx)
+
+	// recalculate limits with destination and current position
+	minX := math.Min(math.Min(currentPose.Point().X, dstPose.Point().X), limits[0].Min)
+	maxX := math.Min(math.Max(currentPose.Point().X, dstPose.Point().X), limits[0].Max)
+	minY := math.Min(math.Min(currentPose.Point().Y, dstPose.Point().Y), limits[1].Min)
+	maxY := math.Min(math.Max(currentPose.Point().Y, dstPose.Point().Y), limits[1].Max)
+	limits = []referenceframe.Limit{
+		{Min: minX, Max: maxX},
+		{Min: minY, Max: maxY},
+		{Min: -2 * math.Pi, Max: 2 * math.Pi},
+	}
+
+	// create a KinematicBase from the componentName
+	baseComponent, ok := ms.components[componentName]
+	if !ok {
+		return false, fmt.Errorf("only Base components are supported for MoveOnMap: could not find an Base named %v", componentName)
+	}
+	kw, ok := baseComponent.(base.KinematicWrappable)
+	if !ok {
+		return false, fmt.Errorf("cannot move base of type %T because it is not KinematicWrappable", baseComponent)
+	}
+	kb, err := kw.WrapWithKinematics(ctx, local, limits)
+	if err != nil {
+		return false, err
+	}
+
+	// check for cancelled context before we are start planning
+	if ctx.Err() != nil {
+		ms.logger.Info("successfully canceled motion service MoveOnGlobe")
+		return true, ctx.Err()
+	}
+
+	// get current position
+	inputs, err := kb.CurrentInputs(ctx)
+	if err != nil {
+		return false, err
+	}
+	seedMap := map[string][]referenceframe.Input{kb.ModelFrame().Name(): inputs}
+
+	// Add the kinematic wheeled base to the framesystem
+	if err := fs.AddFrame(kb.ModelFrame(), fs.World()); err != nil {
+		return false, err
+	}
+
+	// make call to motionplan
+	plan, err := motionplan.PlanMotion(ctx, ms.logger, dstPIF, kb.ModelFrame(), seedMap, fs, wrldst, nil, extra)
+	if err != nil {
+		return false, err
+	}
+
+	// check for cancelled context after we are done planning
+	if ctx.Err() != nil {
+		ms.logger.Info("successfully canceled motion service MoveOnGlobe")
+		return true, ctx.Err()
+	}
+
+	// execute the plan
+	for _, step := range plan {
+		for name, inputs := range step {
+			if len(inputs) == 0 {
```

Comment:
Why check for this? I'm fairly confident that we won't return inputs for things that are not input enabled

Reply:
This I just pasted from what is happening in `MotionService.Move`.
[q]: if it is incorrect to have it here, should it be removed as well from `Move`?

Reply:
we have len(inputs) because an output can look like such:
```
final plan steps: [map[test-base:[{0} {0} {0}] world:[]] map[test-base:[{0} {0} {0}] world:[]] map[test-base:[{0} {0} {0}] world:[]] map[test-base:[{0} {0} {0}] world:[]] map[test-base:[{0} {0} {0}] world:[]] map[test-base:[{0} {0} {0}] world:[]]]
```
we have to account for `world`